### ISSUE TITLE
Using web api response without standard Homey JSON layout

### ISSUE THREAD

**Issue by scanno** [athombv/homey#1746]
**What is your Homey version (Settings → System)?**

1.5.0-RC5

**What did you try to achieve?**

In my owntracks app i am trying to return an array of JSON strings in the HTTP response generated after a HTTP POST request. The body of the HTTP Response is the following:

```
{
        "status": 200,
        "result": [
                {
                        "_type": "location",
                        "tid": "3t",
                        "lat": xxxxcx,
                        "lon": yyyyy,
                        "tst": 1502488223
                },
                {
                        "_type": "location",
                        "tid": "OS",
                        "lat": ccccccc,
                        "lon": yyyyyyy,
                        "tst": 1502218705
                }
        ]
}
```
**But that is not the response that is expected by the client who made the HTTP POST**
The client wants the following payload in the HTTP body:

```
 [
        {
                "_type": "location",
                "tid": "3t",
                "lat": xxxxcx,
                "lon": yyyyy,
                "tst": 1502488223
        },
        {
                "_type": "location",
                "tid": "OS",
                "lat": ccccccc,
                "lon": yyyyyyy,
                "tst": 1502218705
        }
 ]
```

My request is, please allow a way to be able to put a JSON (or other) payload inside the HTTP body without the standard format as it is done now. So without the status and result fields. 

**Any other remarks**

The owntracks app receives HTTP POST requests from owntracks apps on Android or iOS. Those apps have the possibility to show the locations of other people (who are knows by the homey owntracks app) on a map inside the app. This is done by sending the array mentioned above. Right now the clients do not recognize the response because the wanted response is wrapped in the result field.

**Comment by WeeJeWel** [comment]
This will break the API conventions (and therefore all dependent clients) and will not be done. You can additionally spawn your own webserver and serve requests from there.

**Comment by scanno** [comment]
@WeeJeWel spawning a webserver means that the athom cloud can't be used anymore and as such users will need to expose homey to the internet or install a proxy in their network. So that is useless.
I am not saying that the DEFAULT has to be changed. I was asking for a way to override the default structure of the response that is send back. That would not change the default homey API conventions. Unless the result is being processed by your servers...

**Comment by WeeJeWel** [comment]
Yes, that is true. As of Homey v2.0 we have planned to change the result behaviour to only output the `result` property, so `{ status: 200, result: 'foo' }` will become `'foo'`.

**Comment by rickardp** [comment]
Please also consider allowing other types of content than JSON. For example, an additional optional content-type parameter could be used to allow returning plain text etc. This would greatly improve flexibility for app developers. (Sometimes apps do not have the luxury of defining the protocol - it may already have been defined. Opening this up a bit more would help write integration apps)

**Comment by robertklep** [comment]
> For example, an additional optional content-type parameter could be used to allow returning plain text etc.

If this ever were to be implemented, I would opt for using a standard mechanism to implement this (i.e.
 [HTTP content negotiation](https://en.wikipedia.org/wiki/Content_negotiation)).
